src/InputBase.js

```diff
@@ -91,17 +92,20 @@ class InputBase extends React.Component {
   }
 
   renderIcon() {
-    let classes = {
-      'glyphicon': true,
-      'form-control-feedback': true,
-      'glyphicon-ok': this.props.bsStyle === 'success',
-      'glyphicon-warning-sign': this.props.bsStyle === 'warning',
-      'glyphicon-remove': this.props.bsStyle === 'error'
-    };
+    if (this.props.hasFeedback) {
+      if (this.props.feedbackIcon) {
+        return this.props.feedbackIcon;
```

Comment:
@taion: Yeah exactly, so I guess there's two ways to do this. Either
* do as @jquense said: clone the prop, add the class and render it
* add a validation which shows a warning and put the responsibility on the user passing the prop

I'm not sure which way would be preferred though...